api/service/explorer/service.go

```diff
@@ -127,6 +150,108 @@ func (s *Service) GetAddresses(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// UpdateTopAddresses updates 20 top addresses by balance from storage each 10 minutes.
+func (s *Service) UpdateTopAddresses() {
+	if s.GetBalance == nil || s.ReadState == nil {
+		return
+	}
+	ticker := time.NewTicker(time.Minute)
+	prevEpoch := int64(-1)
+	utils.Logger().Info().Msg("start updating top addresses")
+	for range ticker.C {
+		topAddresses := make([]AddrBalance, 0)
+		bytes, err := s.Storage.GetDB().Get([]byte(EpochPrefix), nil)
+		if err != nil {
+			utils.Logger().Error().Err(err).Msg("error reading epoch")
+			continue
+		}
+		epoch := big.NewInt(0)
+		if err = rlp.DecodeBytes(bytes, &epoch); err != nil {
+			utils.Logger().Error().Err(err).Msg("cannot decode epoch")
+			continue
+		}
+		for curEpoch := prevEpoch + 1; curEpoch < epoch.Int64(); curEpoch++ {
```

Comment:
don't need to loop over shard state validator addresses. Just loop through all addresses in the trie.

Reply:
@rlan35 what trie? Which method to use?